src/translation.h

```diff
@@ -30,11 +32,26 @@ extern Translations *g_client_translations;
 class Translations
 {
 public:
-	void loadTranslation(const std::string &data);
+	void loadTranslation(const std::string &filename, const std::string &data);
 	void clear();
 	const std::wstring &getTranslation(
 			const std::wstring &textdomain, const std::wstring &s);
+	const std::wstring &getPluralTranslation(const std::wstring &textdomain,
+			const std::wstring &s, unsigned long int number);
 
 private:
 	std::unordered_map<std::wstring, std::wstring> m_translations;
+	std::unordered_map<std::wstring, std::vector<std::wstring>> m_plural_translations;
+
+	void addTranslation(const std::wstring &textdomain, const std::wstring &original,
+			const std::wstring &translated);
+	void addPluralTranslation(const std::wstring &textdomain,
+			const std::wstring &original,
+			std::vector<std::wstring> &translated);
+	std::wstring unescapeC(const std::wstring &str);
+	void loadPoEntry(const std::map<std::wstring, std::wstring> &entry);
```

Comment:
This is optional: but I feel like there are 3 concerns here mixed into one class. 1) Querying translations, 2) parsing .po, and 3) parsing.tr.

It would be nicer if the parsers were moved into a different class, you could then make a unit test which reads a .po file with various edge cases inside